Server/core/CDynamicLibrary.cpp

```diff
@@ -11,37 +11,28 @@
 
 #include "StdInc.h"
 #include "CDynamicLibrary.h"
-#include "MTAPlatform.h"
 #include "version.h"
 
-#ifdef WIN32
-    #include <stdio.h>
-#endif
-
 void Print(const char* szFormat, ...);
 
 CDynamicLibrary::CDynamicLibrary()
 {
     // Init
-    m_hModule = 0;
-}
-
-CDynamicLibrary::~CDynamicLibrary()
-{
-    // Make sure we unload
-    Unload();
+    m_hModule = nullptr;
 }
 
 bool CDynamicLibrary::Load(const char* szFilename)
 {
     // Unload the previous library
     Unload();
 
-#ifdef WIN32
+#ifdef _WIN32
 
     // Copy the filename
     char szFilenameBuffer[MAX_PATH];
-    strncpy(szFilenameBuffer, szFilename, MAX_PATH);
+    // for "safer" version on windows
+    strncpy_s(szFilenameBuffer, szFilename, MAX_PATH);
+    szFilenameBuffer[MAX_PATH - 1] = '\0';
```

Comment:
I'm pretty sure `strncpy_s` will alway snull terminate it

Reply:
> I'm pretty sure `strncpy_s` will alway snull terminate it

Based on [cppreference](https://en.cppreference.com/w/c/string/byte/strncpy):
> If count is reached before the entire array src was copied, the resulting character array is not null-terminated.

So yeah, null the last character